### PR TITLE
resource_control: add processed_keys and ru to runaway

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? (Required)

Defines `PROCESSED_KEYS` and `REQUEST_UNIT` for runaway queries

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/18575
- Other reference link(s): https://github.com/pingcap/tidb/pull/55147

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by glorv — APPROVED** [review]

File: sql-statements/sql-statement-alter-resource-group.md [hunk 1]
```diff
@@ -53,6 +53,8 @@ ResourceGroupRunawayOptionList ::=
 
 DirectResourceGroupRunawayOption ::=
     "EXEC_ELAPSED" EqOpt stringLit
+|   "PROCESSED_KEYS" EqOpt intLit
+|   "REQUEST_UNIT" EqOpt intLit
```
Comment (HuSharp) [inline sql-statements/sql-statement-alter-resource-group.md:57]: After discussion with PM, we will change `REQUEST_UNIT` to `RU` in sql syntax.
Please refer to this commit to make change, thx~
https://github.com/pingcap/docs-cn/pull/18575/commits/0e20a7de9adcb1e6f75a9400dcff82eb786deacc

File: tidb-resource-control.md [hunk 2]
```diff
@@ -269,9 +269,11 @@ A runaway query is a query (`SELECT` statement only) that consumes more time or
 
 #### `QUERY_LIMIT` parameters
 
-Supported condition setting:
+If a query exceeds any of the following limits, it is identified as a runaway query:
 
-- `EXEC_ELAPSED`: a query is identified as a runaway query when the query execution time exceeds this limit.
+- `EXEC_ELAPSED`: checks whether the query execution time exceeds the limit.
+- `PROCESSED_KEYS`: checks whether the number of keys processed by the Coprocessor exceeds the limit.
+- `REQUEST_UNIT`: checks whether the total number of read and write RUs consumed by the statement exceeds the limit.
```
Comment (HuSharp) [inline tidb-resource-control.md:276]:
```suggestion
- `RU`: checks whether the total number of read and write RUs consumed by the statement exceeds the limit.
```

File: tidb-resource-control.md [hunk 3]
```diff
@@ -296,6 +298,8 @@ The parameters of `QUERY_LIMIT` are as follows:
 | Parameter          | Description            | Note                                  |
 |---------------|--------------|--------------------------------------|
 | `EXEC_ELAPSED`  | When the query execution time exceeds this value, it is identified as a runaway query | EXEC_ELAPSED =`60s` means the query is identified as a runaway query if it takes more than 60 seconds to execute. |
+| `PROCESSED_KEYS` | When the number of keys processed by the Coprocessor exceeds this value, the query is identified as a runaway query | `PROCESSED_KEYS = 1000` means the query is identified as a runaway query if the number of keys processed by the Coprocessor exceeds 1000. |
+| `REQUEST_UNIT`  | When the total number of read and write RUs consumed by the query exceeds this value, this query is identified as a runaway query | `REQUEST_UNIT = 1000` means the query is identified as a runaway query if the total number of read and write RUs consumed by the query exceeds 1000. |
```
Comment (HuSharp) [inline tidb-resource-control.md:302]:
```suggestion
| `RU`  | When the total number of read and write RUs consumed by the query exceeds this value, this query is identified as a runaway query | `RU = 1000` means the query is identified as a runaway query if the total number of read and write RUs consumed by the query exceeds 1000. |
```

File: sql-statements/sql-statement-create-resource-group.md [hunk 4]
```diff
@@ -53,6 +53,8 @@ ResourceGroupRunawayOptionList ::=
 
 DirectResourceGroupRunawayOption ::=
     "EXEC_ELAPSED" EqOpt stringLit
+|   "PROCESSED_KEYS" EqOpt intLit
+|   "REQUEST_UNIT" EqOpt intLit
```
Comment (HuSharp) [inline sql-statements/sql-statement-create-resource-group.md:57]:
```suggestion
|   "RU" EqOpt intLit
```

File: sql-statements/sql-statement-alter-resource-group.md (hunk 1 above)
Comment (HuSharp) [inline sql-statements/sql-statement-alter-resource-group.md:57]:
```suggestion
|   "RU" EqOpt intLit
```

**Review by HuSharp — APPROVED** [review]

**Review by hfxsd — APPROVED** [review]

File: sql-statements/sql-statement-alter-resource-group.md (hunk 1 above)
Comment (lilin90) [reply sql-statements/sql-statement-alter-resource-group.md:57]: Got it.

Comment (lilin90) [reply sql-statements/sql-statement-alter-resource-group.md]: Done in https://github.com/pingcap/docs/pull/18858/commits/7178d520e585cae1414d12e081d196c4d098721f

**Pull request merged**